design-proposals/co-cli.md

```diff
@@ -0,0 +1,99 @@
+# Design Proposal: Command Line Interface for Cluster Orchestration components
+
+Author(s): Julia Okuniewska
+
+Last updated: 2025-05-13
+
+## Abstract
+
+This ADR describes the design of nouns supported by cluster-orchestration
+in command line interface (CLI).
+
+This document is complementary to general CLI design:
+https://github.com/open-edge-platform/edge-manageability-framework/pull/246.
+
+## Proposal
+
+The CLI follows a `verb - noun - subject` pattern.
+
+The target for cluster-orchestration part in CLI is to be able to:
+- create/list/delete clustertemplates
+- mark clustertemplate as default one
+- create cluster with given hosts and given template (or pick default template if not specified)
+- get specified cluster
+- delete/force-delete specified cluster
+- download edgenode's kubeconfig
+
+### Nouns
+- `clustertemplate`
+- `cluster`
+- `en-kubeconfig`
+
+#### Syntax
+Assumption: All CLI commands should include a `--project` or `-p` parameter.
+For simplicity, this parameter is omitted in the commands provided below.
+
+##### clustertemplate
+CRUD operations for clustertemplate:
+```bash
+cli list clustertemplates
+cli list clustertemplates --default
+cli get clustertemplate <name> --version <version>
+cli get clustertemplate versions <name>
+cli apply clustertemplate --file <path_to_clustertemplate.yaml>
+cli delete clustertemplate <name> --version <version>
+```
+Cluster template does not support set/update operation.
+It is immutable.
+
+Extra operations related to clustertemplate:
+```bash
+cli set-default clustertemplate <name> --version <version>
+```
+or if `--version` parameter is not passed, take the latest version for given name.
+
+##### cluster
+CRUD operations for cluster:
+```bash
+cli list clusters
+cli list clusters --summary
+cli get cluster <name>
+cli get cluster details <nodeId>
+cli apply cluster --file <path_to_cluster.yaml>
+cli create cluster \
+    --name <name> \
+    --hosts 6e6422c3-625e-507a-bc8a-bd2330e07e7e:all \ # required in format <uuid:role>
+    --labels key:value \ # optional
+    --template <template name-version> # optional
+cli delete cluster <name>
+cli delete cluster <name> --force
+cli set/update cluster --labels key:value, key2:value2
```

Comment:
is it set or update ?

Reply:
according to general adr - both 
https://github.com/open-edge-platform/edge-manageability-framework/pull/246/files#diff-c189904586748e35b97654c9134bf6fe3bdfbf0d9b141fa942678dd9f47e4ecaR71
![image](https://github.com/user-attachments/assets/eeec11f0-e6bc-414f-8acd-c8556a0ebf21)